PsfLauncher/PsfPowershellScriptRunner.h

```diff
@@ -36,35 +36,94 @@ class PsfPowershellScriptRunner
 
         if (startScriptInformationObject)
         {
-            this->startingScriptInformation = MakeScriptInformation(startScriptInformationObject, stopOnScriptError, currentDirectory);
-            this->startingScriptInformation.doesScriptExistInConfig = true;
+            m_StartingScriptInformation = MakeScriptInformation(startScriptInformationObject, stopOnScriptError, currentDirectory);
+            m_StartingScriptInformation.doesScriptExistInConfig = true;
+            m_AttributeList.Initilize();
         }
 
         if (endScriptInformationObject)
         {
             //Ending script ignores stopOnScriptError.  Keep it the default value
-            this->endingScriptInformation = MakeScriptInformation(endScriptInformationObject, false, currentDirectory);
-            this->endingScriptInformation.doesScriptExistInConfig = true;
+            m_EndingScriptInformation = MakeScriptInformation(endScriptInformationObject, false, currentDirectory);
+            m_EndingScriptInformation.doesScriptExistInConfig = true;
 
             //Ending script ignores this value.  Keep true to make sure
             //script runs on the current thread.
-            this->endingScriptInformation.waitForScriptToFinish = true;
-            this->endingScriptInformation.stopOnScriptError = false;
+            m_EndingScriptInformation.waitForScriptToFinish = true;
+            m_EndingScriptInformation.stopOnScriptError = false;
         }
     }
 
     //RunStartingScript should return an error only if stopOnScriptError is true
     void RunStartingScript()
     {
-        RunScript(this->startingScriptInformation);
+        RunScript(m_StartingScriptInformation, m_AttributeList);
     }
 
     void RunEndingScript()
     {
-        RunScript(this->endingScriptInformation);
+        RunScript(m_EndingScriptInformation, m_AttributeList);
     }
 
 private:
+
+    struct MyProcThreadAttributeList
+    {
+    private:
+        // To make sure the attribute value persists we cache it here.
+        // 0x02 is equivalent to PROCESS_CREATION_DESKTOP_APP_BREAKAWAY_DISABLE_PROCESS_TREE
+        // The documentation can be found here:
+        //https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-updateprocthreadattribute
+        // This attribute tells PSF to
+        // 1. Create Powershell in the same container as PSF.
+        // 2. Any windows Powershell makes will also be in the same container as PSF.
+        // This means that powershell, and any windows it makes, will have the same restrictions as PSF.
+        DWORD createInContainerAttribute = 0x02;
+        std::unique_ptr<LPPROC_THREAD_ATTRIBUTE_LIST> attributeList;

```

Comment:
>LPPROC_THREAD_ATTRIBUTE_LIST [](start = 24, length = 28)

We want a unique_ptr to a struct (not to the pointer to the struct).  Should be,

std::unique_ptr<PROC_THREAD_ATTRIBUTE_LIST> attributeList;